sys/freebsd/evdev.txt

```diff
@@ -0,0 +1,167 @@
+# Copyright 2021 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+# Autogenerated by sys2syz
+
+include <fcntl.h>
+include <dev/evdev/uinput.h>
+include <dev/evdev/input.h>
+
+resource fd_evdev[fd]
+
+openat$evdev(fd const[AT_FDCWD], file ptr[in, string["/dev/evdev"]], flags flags[open_flags], mode const[0]) fd_evdev
+
+ioctl$EVIOCGVERSION(fd fd_evdev, cmd const[EVIOCGVERSION], arg int32)
+ioctl$EVIOCGID(fd fd_evdev, cmd const[EVIOCGID], arg ptr[out, input_id])
+ioctl$EVIOCGREP(fd fd_evdev, cmd const[EVIOCGREP], arg int32)
+ioctl$EVIOCSREP(fd fd_evdev, cmd const[EVIOCSREP], arg int32)
+ioctl$EVIOCGKEYCODE(fd fd_evdev, cmd const[EVIOCGKEYCODE], arg int32)
+ioctl$EVIOCGKEYCODE_V2(fd fd_evdev, cmd const[EVIOCGKEYCODE_V2], arg ptr[inout, input_keymap_entry])
+ioctl$EVIOCSKEYCODE(fd fd_evdev, cmd const[EVIOCSKEYCODE], arg int32)
+ioctl$EVIOCSKEYCODE_V2(fd fd_evdev, cmd const[EVIOCSKEYCODE_V2], arg ptr[in, input_keymap_entry])
+#ioctl$EVIOCGABS(abs)(fd fd_evdev, cmd const[EVIOCGABS(abs)], arg ptr[out, input_absinfo])
```

Comment:
How can I add calls which pass an unknown argument? 
Similar to this, https://github.com/freebsd/freebsd-src/blob/main/sys/dev/evdev/input.h#L90, `len` is unknown so const value is unknown for that ioctl call. How can pass the `len` to it?

Reply:
Good question. Is it possible to do something like

```
define EVIOCGABS_MIN EVIOCGABS(0)
define EVIOCGABS_MAX EVIOCGABS(ABS_MAX)
...
ioctl$EVIOCGABS(fd fd_evdev, cmd intptr[EVIOCGABS_MIN, EVIOCGABS_MAX], ...)
```
?